index.d.ts

```diff
@@ -4,10 +4,10 @@ type NextConfiguration = any;
 
 export = withAwesomeTypescript;
 
-declare function withAwesomeTypescript(
-  awesomeTypescriptOptons?: withAwesomeTypescript.AwesomeTypescriptOptions,
-  nextConfiguration?: NextConfiguration
-): NextConfiguration;
+declare function withAwesomeTypescript(nextOptions: {
+  awesomeTypescriptOptions?: withAwesomeTypescript.AwesomeTypescriptOptions,
+  webpack?(config: NextConfiguration, options: any): NextConfiguration
```

Comment:
There will be an error (at least from some version of typescript) `Object literal may only specify known properties` if you provide something other than this two keys. And personally, I would refrain from making assumptions about `NextConfiguration`, Maybe it's better to rewrite it like this
```
declare function withAwesomeTypescript(
  nextConfig:
    & { awesomeTypescriptOptons?: withAwesomeTypescript.AwesomeTypescriptOptions }
    & NextConfiguration
)
```

Reply:
I just learned typescript. And i found the error in my project. please fix it, thank you